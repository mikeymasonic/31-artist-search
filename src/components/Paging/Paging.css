@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@300&display=swap');

.paging {
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: 'Montserrat', sans-serif;
  color: rgb(93, 93, 93);
  font-size: 1.1vw;
}

.paging button {
  background-color: rgb(133, 220, 226);
  color: white;
  width: 3vw;
  height: 2.6vw;
  font-size: 1vw;
  text-align: center;
  padding: 0;
  font-family: 'Montserrat', sans-serif;
}

.paging button:hover {
  background-color: hotpink;
}

.paging input {
  width: 14vw;
  height: 2vw;
  font-size: 1.4vw;
  font-family: 'Montserrat', sans-serif;
  margin-right: .5vw;
  border: dotted .1vw rgb(197, 197, 197);
  color: rgb(93, 93, 93);
}

.paging p {
  margin: 10px;
}