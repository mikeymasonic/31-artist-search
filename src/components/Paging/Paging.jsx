import React from 'react';
import PropTypes from 'prop-types';

const Paging = ({
  onClickPrevious,
  onClickNext,
  disableNext,
  disablePrev,
  currentPage,
  totalPages
}) => {
  return (
    <div>
      <button onClick={onClickPrevious} disable={disablePrev}>prev</button>
      <p>Page {currentPage} of {totalPages}</p>
      <button onClick={onClickNext} disabled={disableNext}>next</button>
    </div>
  );
};

Paging.propTypes = {
  onClickPrevious: PropTypes.func.isRequired,
  onClickNext: PropTypes.func.isRequired,
  disableNext: PropTypes.bool.isRequired,
  disablePrev: PropTypes.bool.isRequired,
  currentPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired
};

export default Paging;
