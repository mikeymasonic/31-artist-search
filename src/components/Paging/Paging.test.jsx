import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

describe('Paging component', () => {
  it('renders Paging', () => {
    const wrapper = shallow(<Paging onClickPrevious={() => {}} onClickNext={() => {}} disableNext={7} disablePrev={7} currentPage={8} totalPages={4}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
