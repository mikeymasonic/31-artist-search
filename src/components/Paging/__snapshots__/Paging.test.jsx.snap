// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders Paging 1`] = `
<div>
  <button
    disable={7}
    onClick={[Function]}
  >
    prev
  </button>
  <p>
    Page 
    8
     of 
    4
  </p>
  <button
    disabled={7}
    onClick={[Function]}
  >
    next
  </button>
</div>
`;
